test-storybooks/portable-stories-kitchen-sink/react/e2e-tests/component-testing.spec.ts

```diff
@@ -240,4 +237,223 @@ test.describe("component testing", () => {
+    await expect(storyElement).toBeVisible({ timeout: 30000 });
+
+    // Act - Open sidebar context menu and start focused test
+    await page.locator('[data-item-id="addons-group-test"]').hover();
+    await page.locator('[data-item-id="addons-group-test"] div[data-testid="context-menu"] button').click();
+    const sidebarContextMenu = page.getByTestId('tooltip');
+    await sidebarContextMenu.getByLabel('Start Component tests').click();
+
+    // Assert - 5 tests are running and reported
+    await expect(sidebarContextMenu.locator('#testing-module-description')).toContainText('Ran 6 test', { timeout: 30000 });
+    // Assert - 1 failing test shows as a failed status
+    await expect(sidebarContextMenu.getByText('2 stories with errors')).toBeVisible();
+    await expect(sidebarContextMenu.getByLabel('status: failed')).toHaveCount(1);
+
+    await page.click('body');
+    await expect(page.locator('#storybook-explorer-menu').getByRole('status', { name: 'Test status: success' })).toHaveCount(4);
+    await expect(page.locator('#storybook-explorer-menu').getByRole('status', { name: 'Test status: error' })).toHaveCount(2);
+  });
+
+  test("should run focused test for a group", async ({
+    page,
+    browserName,
+  }) => {
+    test.skip(browserName !== "chromium", `Skipping tests for ${browserName}`);
+    // Arrange - Prepare Storybook
+    await setForceFailureFlag(false);
+
+    const sbPage = new SbPage(page, expect);
+    await sbPage.navigateToStory("addons/group/test", "Expected Failure");
+
+    const expandButton = await page.getByLabel('Expand testing module')
+    await expandButton.click();
+
+    const storyElement = sbPage
+      .getCanvasBodyElement()
+      .getByRole("button", { name: "test" });
+    await expect(storyElement).toBeVisible({ timeout: 30000 });
+
+    // Act - Open sidebar context menu and start focused test
+    await page.locator('[data-item-id="addons-group"]').hover();
+    await page.locator('[data-item-id="addons-group"] div[data-testid="context-menu"] button').click();
+    const sidebarContextMenu = page.getByTestId('tooltip');
+    await sidebarContextMenu.getByLabel('Start Component tests').click();
+
+    // Assert - 5 tests are running and reported
+    await expect(sidebarContextMenu.locator('#testing-module-description')).toContainText('Ran 8 test', { timeout: 30000 });
+    // Assert - 1 failing test shows as a failed status
+    await expect(sidebarContextMenu.getByText('3 stories with errors')).toBeVisible();
+    await expect(sidebarContextMenu.getByLabel('status: failed')).toHaveCount(1);
+
+    await page.click('body');
+    await expect(page.locator('#storybook-explorer-menu').getByRole('status', { name: 'Test status: success' })).toHaveCount(4);
+    await expect(page.locator('#storybook-explorer-menu').getByRole('status', { name: 'Test status: error' })).toHaveCount(2);
+  });
+
+  test("should run focused tests without coverage, even when enabled", async ({
```

Comment:
this test is slightly excessive.

It:

1. enables coverage
2. do a focus test
3. assert that coverage is not collected
4. do a full test
5. assert that coverage _is_ collected

I'm not sure if we should keep it or not, you could also argue that it's largely covered by the unit test here:

https://github.com/storybookjs/storybook/pull/29749/files#diff-18fe411ce6b83d11273334c241ee323db41b0c412c66ce04ceef562e5e39a899R172-R201

WDYT @yannbf ?

Reply:
If it's too taxing or flake-prone, I'd say remove it. Seems to be well tested elsewhere, but this E2E test does involve more elements in a real scenario. However I think these elements are tested in the other tests above